src/org/labkey/test/tests/component/GridPanelTest.java

```diff
@@ -1,385 +1,1749 @@
+    // For example if values = ['A', 'B', 'C'] this will return the list ['', 'A', 'B', 'C', 'AB', 'AC', 'ABC', 'BC'].
+    // This is all the sets of the characters from the list, including the empty set.
+    private static List<String> getAllSets(List<String> values, int index)
+    {
+        List<String> allSets = new ArrayList<>();
+        if(index < 0) {
+            allSets.add("");
+            return allSets;
+        }
+
+        allSets = getAllSets(values, index - 1);
+        List<String> newSets = new ArrayList<>();
+
+        for (String allCombination : allSets)
         {
-            sampleSetDataGenerator.addCustomRow(
-                    Map.of("Name", sampleSetDataGenerator.randomString(25),
-                            "descColumn", "used to test issue 39011",
-                            "intColumn", i,
-                            "stringColumn", "filtered_x" + sampleSetDataGenerator.randomString(20)));
+            newSets.add(allCombination + values.get(index));
         }
-        sampleSetDataGenerator.insertRows();
 
-        CoreComponentsTestPage testPage = CoreComponentsTestPage.beginAt(this, getProjectName());
-        QueryGrid grid = testPage.getGridPanel("samples", "filtered_grid_selection_samples");
+        allSets.addAll(newSets);
 
-        // confirm that selectAll is limited to just the records in a filtered set
-        grid.filterColumn("Desc Column", Filter.Operator.EQUAL, "used to test issue 39011");
-        grid.selectAllRows();
-        grid.removeColumnFilter("Desc Column");
-        assertThat(grid.getSelectedRows().size(), is(16));
+        return allSets;
+    }
+
+    /**
+     * Make sure there are no filters or search values persisted for the given sample type.
+     *
+     * @param sampleType The sample type used to populate the grid.
+     * @return A queryGrid object.
+     */
+    public QueryGrid beforeTest(String sampleType)
+    {
+        QueryGrid grid = CoreComponentsTestPage.beginAt(this, getProjectName())
+                .getGridPanel(TEST_SCHEMA, sampleType);
 
+        // Selections can persist, clear them.
         grid.clearAllSelections();
-        assertThat(grid.getSelectedRows().size(), is(0));
 
-        // confirm that search behaves the same way as a filter does
-        grid.search("filtered_x");
-        grid.selectAllOnPage(true);
+        // Searches and filter values shouldn't persist, but clear them just to be safe.
+        grid.clearFilters();
         grid.clearSearch();
-        assertThat(grid.getSelectedRows().size(), is(16));
 
-        // clean up domain after on success
-        sampleSetDataGenerator.deleteDomain(createDefaultConnection());
+        return grid;
+    }
+
+    /**
+     * Helper to see if the paging menu is visible.
+     * Basically this is to check issue 45451.
+     *
+     * @return True if the dropdown page menu is visible, false otherwise.
+     */
+    public boolean isPagingMenuVisible()
```

Comment:
should this helper be on the gird component itself?

Reply:
I wasn't sure if it would be useful outside this test, but I can see that it would make sense to be in a component. I've moved it to the Pager test component.